File: exam-system-java/server/src/main/java/com/yyjy/exam/exam/api/ExamController.java
package com.yyjy.exam.exam.api;

import com.yyjy.exam.common.convention.result.R;
import com.yyjy.exam.entity.exam.entity.ExamRecords;
import com.yyjy.exam.entity.exam.io.req.StartExamReq;
import com.yyjy.exam.entity.exam.io.req.SubmitAnswerReq;
import com.yyjy.exam.exam.service.ExamRecordService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/exams")
@RequiredArgsConstructor
public class ExamController {
	
	private final ExamRecordService examRecordService;
	
	/**
	 * 用户开始考试
	 */
	@PostMapping("/start")
	public R<ExamRecords> startExam(@RequestBody StartExamReq req) {
		return R.ok(examRecordService.startExam(req.paperId(), req.studentName()));
	}
	
	/**
	 * 获取考试记录
	 *
	 * @param id
	 * @return
	 */
	@GetMapping("/{id}")
	public R<ExamRecords> getExamRecord(@PathVariable int id) {
		return R.ok(examRecordService.getExamRecord(id));
	}
	
	@PostMapping("/submit/{examRecordId}")
	public R<Void> submitAnswers(
			@PathVariable int examRecordId,
			@RequestBody List<SubmitAnswerReq> answers
	) {
		examRecordService.submitAnswers(examRecordId, answers);
		return R.ok();
	}
}

File: exam-system-server/server/src/main/kotlin/com/yyjy/controller/ExamController.kt
package com.yyjy.controller

import com.yyjy.common.R
import com.yyjy.models.dto.StartExamDto
import com.yyjy.models.dto.SubmitAnswerDto
import com.yyjy.models.entity.dto.ExamRecordDetail
import com.yyjy.models.entity.dto.ExamStartView
import com.yyjy.service.ExamRecordService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.babyfish.jimmer.client.meta.Api
import org.springframework.web.bind.annotation.*

@Api
@Tag(name = "考试管理", description = "考试流程相关操作，包括开始考试、答题提交、AI批阅、成绩查询等功能")
@RequestMapping("/exams")
@RestController
class ExamController(
    private val examRecordService: ExamRecordService
) {

    @Api
    @Operation(summary = "开始考试")
    @PostMapping("/start")
    fun startExam(@RequestBody req: StartExamDto): R<ExamStartView> {
        return R.ok(examRecordService.startExam(req.paperId, req.studentName))
    }

    @Operation(summary = "查询考试记录详情")
    @GetMapping("/{id}")
    fun getExamRecord(@PathVariable id: Int): R<ExamRecordDetail> {
        return R.ok(examRecordService.getExamRecord(id))
    }

    @Operation(summary = "提交考试答案")
    @PostMapping("/submit/{examRecordId}")
    suspend fun submitAnswers(
        @PathVariable examRecordId: Int,
        @RequestBody answers: List<SubmitAnswerDto>
    ): R<String?> {
        examRecordService.submitAnswers(examRecordId, answers)
        return R.ok(msg="答案已提交")
    }
}